server/src/test/scala/actors/ShiftsActorSpec.scala

```diff
@@ -28,75 +27,250 @@ class ShiftsActorSpec extends CrunchTestLike with ImplicitSender {
+        StaffAssignment("Morning", T1, SDate(s"2017-01-01T09:30:00Z").millisSinceEpoch, SDate(s"2017-01-01T09:44:00Z").millisSinceEpoch, 10, None),
+        StaffAssignment("Morning", T1, SDate(s"2017-01-01T09:45:00Z").millisSinceEpoch, SDate(s"2017-01-01T09:59:00Z").millisSinceEpoch, 10, None),
+        StaffAssignment("Morning", T1, SDate(s"2017-01-01T10:00:00Z").millisSinceEpoch, SDate(s"2017-01-01T10:14:00Z").millisSinceEpoch, 10, None),
+        StaffAssignment("Morning", T1, SDate(s"2017-01-01T10:15:00Z").millisSinceEpoch, SDate(s"2017-01-01T10:29:00Z").millisSinceEpoch, 10, None),
+        StaffAssignment("Morning", T1, SDate(s"2017-01-01T10:30:00Z").millisSinceEpoch, SDate(s"2017-01-01T10:44:00Z").millisSinceEpoch, 10, None),
+        StaffAssignment("Morning", T1, SDate(s"2017-01-01T10:45:00Z").millisSinceEpoch, SDate(s"2017-01-01T10:59:00Z").millisSinceEpoch, 10, None),
+        StaffAssignment("Morning", T1, SDate(s"2017-01-01T11:00:00Z").millisSinceEpoch, SDate(s"2017-01-01T11:14:00Z").millisSinceEpoch, 10, None),
+        StaffAssignment("Morning", T1, SDate(s"2017-01-01T11:15:00Z").millisSinceEpoch, SDate(s"2017-01-01T11:29:00Z").millisSinceEpoch, 10, None),
+        StaffAssignment("Morning", T1, SDate(s"2017-01-01T11:30:00Z").millisSinceEpoch, SDate(s"2017-01-01T11:44:00Z").millisSinceEpoch, 10, None),
+        StaffAssignment("Morning", T1, SDate(s"2017-01-01T11:45:00Z").millisSinceEpoch, SDate(s"2017-01-01T11:59:00Z").millisSinceEpoch, 10, None),
+      ))
       val now: () => SDateLike = () => SDate("2017-01-01T23:59")
       val expireAfterOneDay: () => SDateLike = () => now().addDays(-1)
 
       val actor = system.actorOf(Props(new ShiftsActor(now, expireAfterOneDay, 10)), "shiftsActor")
-
       actor ! UpdateShifts(shifts.assignments)
-      expectMsg(StatusReply.Ack)
+      val response = expectMsgType[ShiftAssignments]
+      val sortedResponse = response.assignments.sortBy(_.start)
+      val sortedExpectedShifts = expectedShifts.assignments.sortBy(_.start)
+      assert(sortedResponse == sortedExpectedShifts)
       actor ! PoisonPill
 
       val newActor = system.actorOf(Props(new ShiftsActor(now, expireAfterOneDay, 10)), "shiftsActor2")
 
       newActor ! GetState
+      val newResponse = expectMsgType[ShiftAssignments]
+      val newSortedResponse = newResponse.assignments.sortBy(_.start)
+      val newSortedExpectedShifts = expectedShifts.assignments.sortBy(_.start)
+      assert(newSortedResponse == newSortedExpectedShifts)
+
+      true
+    }
+
+    "update the shift and check update shift number is applied" in {
+      val startTime = SDate(s"2017-01-01T07:00").millisSinceEpoch
+      val endTime = SDate(s"2017-01-01T12:00").millisSinceEpoch
+
+      val newStartTime = SDate(s"2017-01-01T9:00").millisSinceEpoch
+      val newEndTime = SDate(s"2017-01-01T11:00").millisSinceEpoch
```

Comment:
Same here - reduce the end times so we don't need 40 lines of `StaffAssignment...`